subsys/bluetooth/gatt_dm.c

```diff
@@ -360,15 +395,12 @@ static uint8_t discovery_process_attribute(struct bt_gatt_dm *dm,
 				dm->attrs[0].handle + 1;
 			dm->discover_params.type =
 				BT_GATT_DISCOVER_CHARACTERISTIC;
-			int err = bt_gatt_discover(dm->conn,
-						   &(dm->discover_params));
-
-			if (err) {
-				LOG_ERR("Characteristic discover failed,"
-					" error: %d.",
-					err);
-				discovery_complete_error(dm, err);
-			}
+
+#if defined(CONFIG_BT_GATT_DM_WORKQ_OWN)
+			k_work_submit_to_queue(&bt_gatt_dm_wq, &dm->discover_work);
+#else
+			k_work_submit(&dm->discover_work);
+#endif
```

Comment:
Possible future work: `k_work_submit_to_queue` accepts `NULL` into its `queue` parameter. This makes it use the queue stored in `struct k_work.queue`. It would be nice to establish a pattern of using `NULL` here. It would remove if-defs like this. But we would have to extend the `k_work_init`-function to optionally take a queue to use as default. This makes sense comparing it to an IRQ, where obviously the request raiser does not specify where the handler will run.